packages/std/src/prelude.rs

```diff
@@ -4,3 +4,6 @@ pub use alloc::string::{String, ToString};
 pub use alloc::vec;
 pub use alloc::vec::Vec;
 pub use core::option::Option::{self, None, Some};
+
+#[cfg(not(feature = "std"))]
+core::compile_error!("Please enable `cosmwasm-std`'s `std` feature, as we might move existing functionality to that feature in the future.");
```

Comment:
That place seems a bit strange to me. Why not put it to lib.rs?

Also I'd add this to the message

> Builds without the std feature are currently not expected to work. If you need no_std support see https://github.com/CosmWasm/cosmwasm/issues/1484.